File: app/src/main/java/com/chooloo/www/callmanager/ui/dialpad/DialpadMvpPresenter.java
package com.chooloo.www.callmanager.ui.dialpad;

import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface DialpadMvpPresenter<V extends DialpadMvpView> extends MvpPresenter<V> {
    void onResume();

    void onPause();

    void onKeyClick(int keyCode);

    void onCallClick();

    void onDigitsClick();

    void onDeleteClick();

    void onAddContactClick();

    boolean onLongDeleteClick();

    boolean onLongOneClick();

    boolean onLongZeroClick();

    void onIntentDataChanged(String data);

    void onTextChanged(String text);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/dialpad/DialpadMvpPresenter.kt
package com.chooloo.www.callmanager.ui.dialpad

import com.chooloo.www.callmanager.ui.base.MvpPresenter

interface DialpadMvpPresenter<V : DialpadMvpView> : MvpPresenter<V> {
    fun onKeyClick(keyCode: Int)
    fun onCallClick()
    fun onDeleteClick()
    fun onAddContactClick()
    fun onLongDeleteClick(): Boolean
    fun onLongOneClick(): Boolean
    fun onLongZeroClick(): Boolean
    fun onTextChanged(text: String)
}